commcare_connect/templates/tailwind/components/badges/badge_sm_dropdown.html

```diff
@@ -0,0 +1,73 @@
+{% comment %} {% load static %} {% endcomment %}
+
+<span 
+  x-data="{
+    open: false,
+    dropdownStyle: '',
+    originalOverflow: '',
+    calculatePosition() {
+      const badge = this.$el;
+      const badgeRect = badge.getBoundingClientRect();
+      const viewportHeight = window.innerHeight;
+      const viewportWidth = window.innerWidth;
+      
+      // Calculate available space
+      const spaceBelow = viewportHeight - badgeRect.bottom;
+      const shouldShowAbove = spaceBelow < 200 && badgeRect.top > 200;
+      
+      // Position calculation
+      let top = shouldShowAbove 
+        ? badgeRect.top + window.scrollY - 200
+        : badgeRect.bottom + window.scrollY;
+      
+      // Adjust for right edge
+      let left = Math.min(
+        badgeRect.left, 
+        viewportWidth - 220
+      );
+      
+      this.dropdownStyle = `position: fixed; top: ${top}px; left: ${left}px;`;
+    },
+    toggleDropdown() {
```

Comment:
Can we use alpine based dropdown component described here https://alpinejs.dev/component/dropdown ?